skytemple_files/_resources/patches/asm_patches/happylappy_asm_mods/ai_item_optimizations/common/optimizations_ov29.asm

```diff
@@ -0,0 +1,402 @@
+        b IsHealSeed; If Negative Status, 80%
+Continue:
+    cmp r1,#0x60; Violent Seed
+        beq IsViolentSeed; If below +20 (+10) SPA, 80%. (Never checks ATK)
+    
+    cmp r1,#0x61; Vanish Seed
+        beq IsVanishSeed; If not invisible, 80%.
+    
+    cmp r1,#0x63; Max Elixir
+        beq IsMaxElixir; Scales with PP use: (+30% per empty move slot, +6% for not-filled move slots). Caps at 99%... for some reason
+
+    cmp r1,#0x74; Mix Elixir
+        beq IsMixElixir; If target is a linoone, same as Max Elixir!
+
+    cmp r1,#0x6c; Zinc
+        addeq r4, #0x1; SPDEF is stat 4
+    cmpne r1,#0x66; Iron
+        addeq r4, #0x1; DEF is stat 3
+    cmpne r1,#0x65; Calcium
+        addeq r4, #0x1; SPA is stat 2
+    cmpne r1,#0x64; Protein
+        addeq r4, #0x1a; Stat offset is 0x1a (+1 for each subsequent stat!)  
+        beq VitaminCheck;  If STAT < 250, 100%.
+
+    cmp r1,#0x68; DropEye Seed
+        beq IsDropEye; If Throwing and Not Drop-Eyed, 80%.
+
+    cmp r1,#0x6a; Slip Seed
+        beq IsSlipSeed; If mobility type isn't 0 or 4, and not slipping: If Adjacent to Enemy, 100%, else 10%.
+
+    cmp r1,#0x6d; Apple
+    cmpne r1,#0x6e; Big Apple
+    cmpne r1,#0x70; Huge Apple
+        beq IsApple; If Belly < 10, 100%.
+
+    cmp r1,#0x6f; Grimy Food
+        beq odds_30; IN GENERAL, 30%. 
+     
+    cmp r1,#0x76; Dough Seed
+        beq IsDoughSeed; If on the team, 100%. (In other words, wild dungeon pokemon cannot use this!)
+
+    cmp r1,#0x89; Gravelyrock
+        beq IsGravelyrock; If Bonsly or Sudowoodo, If thrown 70%, Used 100%.
+    
+    cmp r1,#0xa7; Gone Pebble
+        beq IsGonePebble; If not enduring, and adjacent to an enemy, 80%.
+
+    eor r1,#0x100; If 0x14F, becomes 0x4F.
+    cmp r1,#0x4F; Rollcall Orb
+        cmpeq r5,#0x0; If using...    
+            moveq r0,#20; Add 20%   
+            beq exit; Exit with whatever % we have!
+    ; See the Optimizations patch for info on how to better set up item IDs above 0xFF;
+    ; Any item NOT listed above will never be used by the AI.
+odds_0:
+    mov r0, #0; reset it just to prevent random 1% use chances...
+    b exit;   
+
+
+ComputeHunger:
+; Returns hunger in r4, leaves r0 as it was found!
+    push r0,r2,r14;
+    add r0, r6, #0x100
+    ldrh r1, [r0, #0x46]
+    sub r2, sp, #4
+    strh r1, [r2]
+    ldrh r0, [r0, #0x48]
+    strh r0, [r2, #2]
+    ldr r0, [r2]
+    bl CeilFixedPoint
+    mov r4, r0;
+    pop r0,r2,r15;
+
+IsDietRibbon:
+    cmp r5,#0x0; If using...
+    beq odds_0;
+IsHungerSeed:
+    bl ComputeHunger; Custom function above. Returns hunger in r4.
+    cmp r4,#0
+    beq odds_0;
+    b odds_50;  
+ 
+IsWhifforNoAim:
+    cmp r5,#0x1; If throwing... 
+True_50_False_0:
+    beq odds_50;    
+    b odds_0;
+
+; These lines are stupidly common. Might as well make a proper function!
+Adjacent2Enemy:
+    push r14;
+    mov r0,r7; 
+    bl IsAdjacentToEnemy
+    cmp r0,#0x1;
+    mov r0,#0;
```

Comment:
Why the unconditional mov? I see you're effectively returning the result of the comparison in the status register (cursed), but in that case, you don't need to set r0, right?

Edit: Oh, it's because you might jump to something like odds_50. You might want to add a comment here, it's not very obvious.